acceptance-tests/helpers/s3vpcendpoint/main.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+)
+
+type createParams struct {
+	allowedVPCID   string
+	allowedUserARN string
+}
+
+func main() {
+	createCommand := flag.NewFlagSet("create", flag.ExitOnError)
+	deleteCommand := flag.NewFlagSet("delete", flag.ExitOnError)
+
+	if len(os.Args) < 2 {
+		fmt.Println("expected 'create' or 'delete' subcommands")
+		os.Exit(1)
+	}
+
+	// Environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are required
+	svc := createAWSClientOrPanic()
+
+	switch os.Args[1] {
+	case "create":
+		params := getCreateParamsFromArgsOrFail(createCommand)
+
+		vpcEndpointID := createVPCEndpoint(svc, params)
+		// vpcEndpointID to the stdout to be able to pipe it to the next step
+		fmt.Println(vpcEndpointID)
+	case "delete":
+		vpcEndpointID := getVPCEndpointIDFromArgsOrFail(deleteCommand)
+		deleteVPCEndpoint(svc, vpcEndpointID)
+	default:
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+}
+
+func getVPCEndpointIDFromArgsOrFail(deleteCommand *flag.FlagSet) string {
+	err := deleteCommand.Parse(os.Args[2:])
+	if err != nil {
+		fmt.Printf("error parsing 'delete' subcommand %s\n", err)
+		os.Exit(1)
+	}
+
+	if len(deleteCommand.Args()) != 1 {
+		fmt.Println("expected VPC Endpoint ID argument for 'delete' subcommand")
+		os.Exit(1)
+	}
+
+	vpcEndpointID := deleteCommand.Args()[0]
+	return vpcEndpointID
+}
+
+func getCreateParamsFromArgsOrFail(createCommand *flag.FlagSet) createParams {
+	err := createCommand.Parse(os.Args[2:])
+	if err != nil {
+		fmt.Printf("error parsing 'create' subcommand %s\n", err)
+		os.Exit(1)
+	}
+
+	if len(createCommand.Args()) != 2 {
+		fmt.Println("expected allowed VPC ID argument for 'create' subcommand")
+		os.Exit(1)
+	}
+
+	return createParams{
+		allowedVPCID:   createCommand.Args()[0],
+		allowedUserARN: createCommand.Args()[1],
+	}
+}
+
+func createAWSClientOrPanic() *ec2.Client {
+	accessKey := os.Getenv("AWS_ACCESS_KEY_ID")
+	secretKey := os.Getenv("AWS_SECRET_ACCESS_KEY")
+
+	if accessKey == "" || secretKey == "" {
+		panic("Environment variables AWS_ACCESS_KEY_ID and/or AWS_SECRET_ACCESS_KEY are not set")
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
```

Comment:
Using the client is fine. In other tests we use the AWS CLI: https://github.com/cloudfoundry/csb-brokerpak-aws/blob/main/acceptance-tests/helpers/dms/aws.go
It could be easier to do this instead. Although in that case I'd suggest factoring this out from the DMS package.